cedar-policy-validator/src/schema_file_format.rs

```diff
@@ -92,19 +142,45 @@ impl SchemaFragment {
 #[cfg_attr(feature = "wasm", tsify(into_wasm_abi, from_wasm_abi))]
 pub struct NamespaceDefinition {
     #[serde(default)]
-    #[serde(with = "::serde_with::rust::maps_duplicate_key_is_error")]
+    #[serde(deserialize_with = "deserialize_common_types")]
     #[serde(rename = "commonTypes")]
-    pub common_types: HashMap<SmolStr, SchemaType>,
+    pub common_types: HashMap<Id, SchemaType>,
     #[serde(rename = "entityTypes")]
-    #[serde(with = "::serde_with::rust::maps_duplicate_key_is_error")]
-    pub entity_types: HashMap<SmolStr, EntityType>,
+    #[serde(deserialize_with = "deserialize_entity_types")]
+    pub entity_types: HashMap<Id, EntityType>,
     #[serde(with = "::serde_with::rust::maps_duplicate_key_is_error")]
     pub actions: HashMap<SmolStr, ActionType>,
 }
 
+fn deserialize_common_types<'de, D>(
+    deserializer: D,
+) -> std::result::Result<HashMap<Id, SchemaType>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    deserialize_hash_map(
+        |key| Id::from_normalized_str(&key),
```

Comment:
Can you confirm that we have tests checking that this and all similar parsing checks that you've edited are correctly applied? I imagine most are tested, but I'd like to be sure of it.

Reply:
- [ ] duplicate keys
- [ ] namespace id
- [ ] entity type id
- [ ] common type id
- [ ] entity type reference
- [ ] common type reference